capsules/extra/src/test/aes.rs

```diff
@@ -56,13 +59,15 @@ impl<'a, A: AES128<'a> + AES128ECB> TestAes128Ecb<'a, A> {
         key: &'a mut [u8],
         source: &'static mut [u8],
         data: &'static mut [u8],
+        test_decrypt: bool,
```

Comment:
This seems to go against a lot of the static checks of Rust. The AES support is used in the kernel and exposed to userspace as well. It seems really clunky that we just return `INVAL` when trying to decrypt data.

I think this should be split in the HIL, Then all users (tests, kernel and apps) don't try to perform operations the hardware doesn't support. That way it is also known at compile time instead of getting a `INVAL` when trying to perform an operation

Reply:
That might make sense, but it is still helpful to be able to test nrf52 encrypt in the meantime.

Reply:
This is making it harder to update it in the future though and diverging testing from actual use cases.

This quick fix is digging us a deeper hole

Reply:
I mean it's just a single boolean in a test.

I'm not convinced we want to rewrite the HIL to support a single chip which has very limited AES support (it only supports use cases needed for the radio). It seems like very much the exception.

Reply:
If the nRF is the exception maybe it makes more sense to create a `nrf_aes_test.rs` then with no decryption. That way it's clear that this is an exception and a weird one off

Reply:
Um... well... I think the one boolean (and associated logic in the test) is a reasonable amount of tech debt, but also the trait implementations for the NRF5x AES are a kludge...

Perhaps just adding a small comment near the function argument here that this is only to support an idiosyncrasy of the nrf chip's AES driver not supporting decrypt, and should be removed if/when that's resolved.

But i don't think we either block this on fixing the nrf5x implementation or have a totally separate and otherwise duplicate test for the nrf